bootstrap/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -250,15 +252,115 @@ func (kustomize *kustomize) initK8sClients() error {
 	return nil
 }
 
+func (kustomize *kustomize) deleteFlags(usage string) *kubectldelete.DeleteFlags {
+	cascade := true
+	gracePeriod := -1
+	// setup command defaults
+	all := false
+	force := false
+	ignoreNotFound := false
+	now := false
+	output := ""
+	labelSelector := ""
+	fieldSelector := ""
+	timeout := time.Duration(0)
+	wait := true
+	filenames := []string{"-"}
+	recursive := false
+	return &kubectldelete.DeleteFlags{
+		FileNameFlags:  &genericclioptions.FileNameFlags{Usage: usage, Filenames: &filenames, Recursive: &recursive},
+		LabelSelector:  &labelSelector,
+		FieldSelector:  &fieldSelector,
+		Cascade:        &cascade,
+		GracePeriod:    &gracePeriod,
+		All:            &all,
+		Force:          &force,
+		IgnoreNotFound: &ignoreNotFound,
+		Now:            &now,
+		Timeout:        &timeout,
+		Wait:           &wait,
+		Output:         &output,
+	}
+}
+
+func (kustomize *kustomize) initApplyOptions(o *kubectlapply.ApplyOptions, f cmdutil.Factory) error {
+	var err error
+	// allow for a success message operation to be specified at print time
+	o.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
+		o.PrintFlags.NamePrintFlags.Operation = operation
+		if o.DryRun {
+			o.PrintFlags.Complete("%s (dry run)")
+		}
+		if o.ServerDryRun {
+			o.PrintFlags.Complete("%s (server dry run)")
+		}
+		return o.PrintFlags.ToPrinter()
+	}
+	o.DiscoveryClient, err = f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	o.DeleteOptions = o.DeleteFlags.ToOptions(dynamicClient, o.IOStreams)
+	o.ShouldIncludeUninitialized = false
+	o.OpenApiPatch = true
+	o.OpenAPISchema, _ = f.OpenAPISchema()
+	o.Validator, err = f.Validator(false)
+	o.Builder = f.NewBuilder()
+	o.Mapper, err = f.ToRESTMapper()
+	if err != nil {
+		return err
+	}
+	o.DynamicClient, err = f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (kustomize *kustomize) tempFile(data []byte) *os.File {
+	tmpfile, err := ioutil.TempFile("/tmp", "kout")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if _, err := tmpfile.Write(data); err != nil {
+		log.Fatal(err)
+	}
+	if _, err := tmpfile.Seek(0, 0); err != nil {
+		log.Fatal(err)
+	}
+	return tmpfile
+}
+
+func (kustomize *kustomize) cleanup(tmpfile *os.File, stdin *os.File) error {
+	os.Stdin = stdin
+	if err := tmpfile.Close(); err != nil {
+		log.Fatal(err)
+	}
+	return os.Remove(tmpfile.Name())
+}
+
 // Apply deploys kustomize generated resources to the kubenetes api server
 func (kustomize *kustomize) Apply(resources kftypesv3.ResourceEnum) error {
-	if err := kustomize.initK8sClients(); err != nil {
+	os.Setenv(envBackoffBase, "1")
```

Comment:
Why are we setting these environment variables? Doesn't this prevent users from customizing the behavior by setting these environment variables?

Reply:
yes, you're right. I don't think I need them. The 'magic' setting is ShouldIncludeUninitialized=false which means wait for creation for new objects